src/modules/shared/pages/innovation/sections/section-summary.component.ts

```diff
@@ -0,0 +1,207 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { ActivatedRoute, NavigationEnd } from '@angular/router';
+import { Subscription, forkJoin, of } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import { CoreComponent } from '@app/base';
+import { ContextInnovationType } from '@app/base/types';
+
+import { WizardEngineModel, WizardSummaryType } from '@modules/shared/forms';
+import { InnovationDocumentsListOutDTO, InnovationDocumentsService } from '@modules/shared/services/innovation-documents.service';
+import { INNOVATION_SECTION_STATUS, InnovationSectionEnum, InnovationStatusEnum } from '@modules/stores/innovation';
+import { getInnovationRecordConfig, innovationSectionsWithFiles } from '@modules/stores/innovation/innovation-record/ir-versions.config';
+import { InnovationSectionStepLabels, InnovationSectionsVersions } from '@modules/stores/innovation/innovation-record/ir-versions.types';
+
+
+
+@Component({
+    selector: 'shared-innovation-summary',
+    templateUrl: './section-summary.component.html'
+})
+export class InnovationSectionSummaryComponent extends CoreComponent implements OnInit, OnChanges {
+
+  @Input() innovation!: ContextInnovationType;
+  @Input() sectionId: string | undefined;
+
+  sectionInfo: {
+    id: string,
+    nextSectionId: null | string,
+    title: string,
+    status: { id: keyof typeof INNOVATION_SECTION_STATUS, label: string },
+    submitButton: { show: boolean, label: string },
+    isNotStarted: boolean,
+    hasEvidences: boolean,
+    wizard: WizardEngineModel,
+    allStepsList: InnovationSectionStepLabels,
+    date: string,
+    submittedBy: null | { name: string, isOwner?: boolean },
+    openTasksCount: number
+  };
+
+  sectionSubmittedText: string = '';
+
+  sectionsIdsList: string[];
+  summaryList: WizardSummaryType[] = [];
+  evidencesList: WizardSummaryType[] = [];
+  documentsList: InnovationDocumentsListOutDTO['data'] = [];
+
+  baseUrl: string;
+  isSectionDetailsPage: string | undefined = undefined;
+
+
+  // Flags
+  isInnovatorType: boolean;
+  isAccessorType: boolean;
+  isAssessmentType: boolean;
+  shouldShowDocuments = false;
+
+  paramSubscription: Subscription = new Subscription();
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private innovationDocumentsService: InnovationDocumentsService
+  ) {
+
+    super();
+
+    this.innovation = this.innovation ?? this.stores.context.getInnovation();
```

Comment:
Duplication of code

Reply:
I think you don't need to check "this.innovation ?? this.stores.context.getInnovation();" because you'll always receive the innovation. Am I wrong?